### DISCUSSION TITLE
Len_scale question

### DISCUSSION THREAD

**Discussion by fcarrillo-brenes** [GeoStat-Framework/GSTools#233] [Q&A] [answered]
Hi, I have a  question about the parameter 
len_scale in the model and in the variogram fit. 
I'm using the library and I have a doubt about this value, when I don't set it in a fixed value, it changes when I run the code. Otherwise, when I fix it the result changes but the values of the parameters don't change every run.
Someone can help me to understand what is the meaning of the parameter and its function in the variogram fit. I will appreciate it so much.

**Comment by LSchueler** [comment, marked as answer]
Hi FcarrilloBre17,

I don't quite understand your question. I think the best thing for you to do is to provide a [minimal working example](https://stackoverflow.com/help/minimal-reproducible-example). Then we can have a look at the problems you are facing. This is a complete shot in the dark, but could it be that you are not fixing your seed for the pseudo random number generation?

And did you have a look at the documentation?
This is how [the covariance model is defined](https://geostat-framework.readthedocs.io/projects/gstools/en/latest/examples/02_cov_model/index.html) and how the `len_scale` is used there.
[This](https://geostat-framework.readthedocs.io/projects/gstools/en/latest/generated/gstools.variogram.vario_estimate.html) is how the variogram estimation works.
[Here](https://geostat-framework.readthedocs.io/projects/gstools/en/latest/examples/03_variogram/index.html) are some examples of how to use the variagram estimation.

**Reply by fcarrillo-brenes** [reply]
Hi, 
well I used a fixed seed in my program, my question was more about the meaning of thelen_scale parameter, seeing some examples related to coordinates you fixed the values for len_scale but I don't get it why. 
And that was the reason of posting this question.

**Reply by LSchueler** [reply]
What exactly is your question regarding the meaning of the length scale? Have a look at the definition of the covariance model I linked above. It basically tells you in what distance you can still expect significant covariances between two points. If that does not help, please formulate your question a bit more precisely.
In some examples we generate spatial random fields without estimating the variogram (or covariance model). Thus, we fix the length scale to a certain value. In a real workflow you would first estimate it from data (one possibility would be variogram estimation) and then use that value to generate spatial random fields.
However, you do not always generate spatial random fields from the results of a variogram estimation.

[Here](https://geostat-framework.readthedocs.io/projects/gstools/en/latest/examples/02_cov_model/04_different_scales.html#sphx-glr-examples-02-cov-model-04-different-scales-py) are some different definitions of length scales you could also use.

**Reply by fcarrillo-brenes** [reply]
Your last answer has solved my doubt, sorry if I was not more precise with the question. 
And Thank you for the answer
